product_attribute_model_link/models/product_attribute_value.py

```diff
@@ -0,0 +1,184 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductAttributeValue(models.Model):
+    _inherit = "product.attribute.value"
+
+    linked_record_ref = fields.Reference(
+        selection="_selection_target_model", string="Linked record"
+    )
+
+    @api.model
+    def _selection_target_model(self):
+        models = self.env["ir.model"].search([])
+        return [(model.model, model.name) for model in models]
+
+    def convert_attribute_value(self, value, linked_field):
+        """
+        Convert attribute value name to the appropriate data
+        type based on the linked field's type.
+
+        :param value: Value to be converted.
+        :return: Converted value or the original value if no conversion is needed.
+        """
+        if linked_field and linked_field.ttype in ["float", "integer"]:
+            return float(value) if linked_field.ttype == "float" else int(value)
+        return value
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        """
+        Override to create records based on attribute values.
+
+        :param vals_list: List of dictionaries containing attribute values for creating records.
+        :return: Created records.
+        :raise: ValidationError if conversion fails for a field.
+        """
+        for vals in vals_list:
+            attribute_id = vals.get("attribute_id")
+            product_attribute = self.env["product.attribute"].browse(attribute_id)
+            # Create new record based on attribute value if create_from_attribute_values is True
+            if (
+                product_attribute.create_from_attribute_values
```

Comment:
Could you add check exists `product_attribute` variable in this condition like this:
```py
if product_attribute and product_attribute.create_from_attribute_values and not self.env.context.get("operation_from_record"):
    ...
```